core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -249,6 +266,15 @@ private ClientTransport obtainActiveTransport() {
         // start() must return before we set activeTransport, since activeTransport is accessed
         // without a lock.
         activeTransport = newActiveTransport;
+        backoff = null;
+      } else {
```

Comment:
Since "Implementations must not call listener from within start()," this should generally not happen. At the very least it is not a preferred signal of a connection failure, in part because start() should not block on I/O. Looking through the code, no transport calls transportShutdown within start().